### ISSUE TITLE
Question about the strategy for saving the best model results

### ISSUE THREAD

**Issue by ZhangVerse** [ZzoomD/FairINV#1]
Hello,

Thank you for sharing your excellent work! I find the motivation behind your paper very inspiring and the proposed methodology intriguing.
But I have a question regarding your approach to saving the best model results:
Could you share more details on how you determined the strategy for saving the best-performing model?
Understanding your thought process and implementation would greatly help in replicating and building upon your work.

Thank you in advance for your time and response!

**Comment by ZzoomD** [comment]
Hello, ZhangVerse

Thank you for your question. 
We save the best models based on their optimal performance (in terms of utility-fairness or loss) on the validation dataset. For more details, please refer to the train_model function in the FairINV class within the "[fairinv.py](https://github.com/ZzoomD/FairINV/blob/master/fairinv.py)" file. Our approach to selecting the best model is inspired by the methodologies used in FairVGNN (https://github.com/yuwvandy/FairVGNN/blob/main/fairvgnn.py) and NIFTY (https://github.com/chirag-agarwall/nifty/blob/main/nifty_sota_gnn.py).

**Comment by ZhangVerse** [comment]
> Hello, ZhangVerse
> 
> Thank you for your question. We save the best models based on their optimal performance (in terms of utility-fairness or loss) on the validation dataset. For more details, please refer to the train_model function in the FairINV class within the "[fairinv.py](https://github.com/ZzoomD/FairINV/blob/master/fairinv.py)" file. Our approach to selecting the best model is inspired by the methodologies used in FairVGNN (https://github.com/yuwvandy/FairVGNN/blob/main/fairvgnn.py) and NIFTY (https://github.com/chirag-agarwall/nifty/blob/main/nifty_sota_gnn.py).

Thank you so much for your previous reply! Your project has been incredibly helpful to me, and I deeply appreciate it.

I currently have a new question while using the code: why do we need to set the label_number parameter in the load_data function instead of selecting all the data? Is there a specific consideration behind this design, such as performance optimization or something related to the model's training effectiveness?

I would be extremely grateful if you could take the time to answer my question!

**Comment by ZzoomD** [comment]
Hello, ZhangVerse

The purpose of the label_number parameter is to set the total number of samples in the training and validation sets combined. You can use the default setting we provide, or you can customize the value of label_number yourself.

**Comment by ZhangVerse** [comment]
Thanks, I understand now. Using a small amount of training data to achieve improvements in utility and fairness demonstrates that the model design is excellent! Great work!
